src/Tomcat/TomcatController.ts

```diff
@@ -19,74 +17,65 @@ export class TomcatController {
     constructor(private _tomcatModel: TomcatModel, private _extensionPath: string) {
     }
 
-    public getTomcatServer(serverName: string): TomcatServer {
-        return this._tomcatModel.getTomcatServer(serverName);
-    }
-
-    public getServerSet(): TomcatServer[] {
-        return this._tomcatModel.getServerSet();
-    }
-
     public async deleteServer(tomcatServer: TomcatServer): Promise<void> {
-        if (!tomcatServer) {
-            throw (new Error(DialogMessage.noServer));
-        }
-
-        if (tomcatServer.isStarted()) {
-            const confirmation: MessageItem = await vscode.window.showWarningMessage(DialogMessage.deleteConfirm, DialogMessage.yes, DialogMessage.cancel);
-            if (confirmation !== DialogMessage.yes) {
-                return;
+        const server: TomcatServer = await this.precheck(tomcatServer);
```

Comment:
This changelist contains too many changes: +262 -340, 602 in total. It makes the review process hard and slow. Both the reviewers and the reviewees will suffer.
In the future, please try to focus on one thing at a time following Single Responsibility Principle.  The recommend lines of changes are less than 200 in a pull request.

Reply:
Yeah, will do a more widely communication next time if any big change